### ISSUE TITLE
rolled back releases not removed from .deb/releases

### ISSUE THREAD

**Issue by arderyp** [deployphp/deployer#2134]
**Describe the bug**
When I roll back a broken release, the relase symbolic link is removed, and the appropriate subdirectory is removes from `releases/`.  However, the release is not removed from `.dep/releases` (the last line).  I've got a custom task that handles this, but I am wondering if I'm missing something.  Is the `.dep/releases` file not updated on rollback for a reason?

**To reproduce**
Deploy, `cat .dep/releases`, rollback, `cat .dep/releases`

**Expected behavior**
You tell me.

**Screenshots**

**Environment**
* Deployer version: 6.8.0
* PHP version: 7.4.9
* Deployment target(s) OS: linux

**Content of `deploy.php`**
```php
// your code here
```

**Output log with enabled option for verbose output `-vvv`**
```sh
# output here
```

**Additional context**

**Comment by antonmedv** [comment]
There is no need to update .dep/releases. Think of it as immutable history of releases.

**Comment by arderyp** [comment]
but, if I roll back, and physically remove a release from `ROOT/releases`, a pointer to the removed release still exists in `.dep/releases`.  Is that not problematic?  If so, I guess I'll stop removing the lines.

**Comment by arderyp** [comment]
I havent tested int, but would this be an issue if you have, for example:

```
#.dep/releases
20200813111158,1
20200813131448,2
20200813174301,3
20200814132820,4
20200814133511,5
20200814134119,6
```
Then you roll back `6`, then you deploy and now have:

```
#.dep/releases
20200813111158,1
20200813131448,2
20200813174301,3
20200814132820,4
20200814133511,5
20200814134119,6  # this release is missing because we rolled it back, which removed it from ROOT/releases
20200814173239,7
```

Then you try to roll back 7, and deployer tries to install 6.  Wouldn't that cause an issue?

**Comment by antonmedv** [comment]
It’s for logging and understanding what is next release version should be. What is you remove all releases?

**Comment by arderyp** [comment]
@antonmedv I don't fully understand the question.  What if I remove all releases, and all lines from `.deb/releases`?  I don't see a problem.  

If there is nothing in `.deb/releases` and there are no actual releases in `releases/` its basically like a brand new system in its initial state.   In other words, the same situation as when you are setting up your host for the first time.  You'd just run `dep deploy {HOST}`, and I believe it would handle that fine without issue.  Am I missing something?

**Comment by antonmedv** [comment]
Imagine this scenario :
* Deploy 5 buggy releases.
* Manually ssh to server and delete all releases/*
* Start new release
* There is .dep/releases -> correct release number is chosen.

**Comment by arderyp** [comment]
What do you mean by "release number" when you say "correct release number is chosen"?  As far as I see it, the actual numbers in `releases/` are meaningless, like an index column in a database.  The more meaningful number is the actual versioned git tag (example `dep deploy production --tag=1.2.3`.  

Are you just saying that, in your scenario the system would create `releases/6` instead of `releases/1`?  If so, I again don't see a problem.  Maybe I'm misunderstanding?

**Comment by antonmedv** [comment]
Yes, it will create 6. This feature was created by some users who encountered issues. This is realy important in sense of server cache systems.
